### PR TITLE
[MCH-144-45] multicall

### PR DESCRIPTION
**Pull request by poolpitako**

### CODE REVIEW

File: packages/yswaps/test/e2e/TradeFactory/tarde-with-multicall-async-swapper.spec.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+import moment from 'moment';
+import { erc20, evm } from '@test-utils';
+import * as fixtures from '../../fixtures';
+import { contract, given, then } from '@test-utils/bdd';
+import { expect } from 'chai';
+import { IERC20, TradeFactory } from '@typechained';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signers';
+import { mergeTransactions } from '@scripts/libraries/multicall';
+
+contract('MultiCallSwapper', () => {
```
Comment (b0dhidharma) [inline packages/yswaps/test/e2e/TradeFactory/tarde-with-multicall-async-swapper.spec.ts:12]: I'll probably extend / modify this test later 😁

File: packages/yswaps/test/fixtures.ts [hunk 2]
```diff
@@ -147,3 +147,49 @@ export const uniswapV2SwapperFixture = async (
...
+    otcPoolGovernor
+  );
+  const MultiSendCallOnlyFactory = await ethers.getContractFactory('contracts/libraries/MultiSendCallOnly.sol:MultiSendCallOnly');
+  const multiCallAsyncSwapperFactory = await ethers.getContractFactory('contracts/swappers/async/MultiCallSwapper.sol:MultiCallSwapper');
+  const multiCallOptimizedAsyncSwapperFactory = await ethers.getContractFactory(
+    'contracts/swappers/async/MultiCallOptimizedSwapper.sol:MultiCallOptimizedSwapper'
+  );
+  const multiCallOptimizedAsyncSwapper = await multiCallOptimizedAsyncSwapperFactory.deploy(masterAdmin, tradeFactory.address);
+  const owner = await wallet.generateRandom();
+  await ethers.provider.send('hardhat_setBalance', [owner.address, utils.parseEther('10').toHexString()]);
```
Comment (b0dhidharma) [inline packages/yswaps/test/fixtures.ts:184]: Why ? Not being used anywhere afaik.

**Review by b0dhidharma — COMMENTED** [review]
Overall, looks good - although Im not sold on `MultiCall` solution. Will sync up offline. I know how much working on `typescript` must have hurt @poolpitako, so THANKS for this hard work. 🚀

File: packages/yswaps/hardhat.config.ts [hunk 3]
```diff
@@ -1,15 +1,15 @@
 import hardhatConfig from '../commons/hardhat.config';
 
 const config = hardhatConfig({
-  networks: ['mainnet', 'ropsten', 'polygon', 'fantom'],
+  networks: ['mainnet'],
```
[the remark this answers is not in the document]
Comment (skeletor-spaceman) [reply packages/yswaps/hardhat.config.ts:4]: reverting this change. We need a fix for unused networks blocking execution

**Pull request closed without merging**